lib/Sema/CSWalker.cpp

```diff
@@ -205,6 +205,67 @@ class LocalCSWalker : public glu::ast::ASTWalker<LocalCSWalker, void> {
             )
         );
     }
+
+    void postVisitUnaryOpExpr(glu::ast::UnaryOpExpr *node)
+    {
+        auto *operandTy = node->getOperand()->getType();
+        auto *resultTy = node->getType();
+
+        auto &arena = node->getModule()->getContext()->getTypesMemoryArena();
+
+        auto *expectedFunctionTy = arena.create<glu::types::FunctionTy>(
+            llvm::ArrayRef<glu::types::TypeBase *>(&operandTy, 1), resultTy
```

Comment:
```suggestion
            { operandTy }, resultTy
```
just use an initializer list...

Reply:
the problem is that this is seriously hideous, you create a one-dimensional array from a variable on the stack
```suggestion
            std::initializer_list { operandTy }, resultTy
```
maybe? or
```suggestion
            llvm::ArrayRef<glu::types::TypeBase *> { operandTy }, resultTy
```